### ISSUE TITLE
在Deforming Plate 数据集中，网格是动态变化的，AMG是如何处理这种动态变化网格的

### ISSUE THREAD

**Issue by SereinQAQ** [lizhihao2022/AMG#2]

**Comment by lizhihao2022** [comment]
```python
def single_process(self, data, in_t, out_t, duration_t, mode='train'):
        stress = torch.from_numpy(data['stress'])
        world_pos = torch.from_numpy(data['world_pos'])
        node_type = torch.from_numpy(data['node_type'][0])
        mesh_pos = torch.from_numpy(data['mesh_pos'][0])
        cell = torch.from_numpy(data['cells'][0])
        
        if mode == 'train':
            x = stress[:in_t, :]
            y = stress[in_t:in_t+1, :]
            x_world = world_pos[:in_t, :]
            for i in range(1, duration_t):
                x = torch.cat([x, stress[i:in_t+i, :]], dim=0)
                y = torch.cat([y, stress[in_t+i:in_t+i+1, :]], dim=0)
                x_world = torch.cat([x_world, world_pos[i:in_t+i, :]], dim=0)
        else:
            x = stress[out_t-in_t:out_t, :]
            y = stress[out_t:out_t+1, :]
            x_world = world_pos[out_t-in_t:out_t, :]
            for i in range(1, duration_t):
                x = torch.cat([x, stress[out_t+i-in_t:out_t+i, :]], dim=0)
                y = torch.cat([y, stress[out_t+i:out_t+i+1, :]], dim=0)
                x_world = torch.cat([x_world, world_pos[out_t+i-in_t:out_t+i, :]], dim=0)
        
        all_data = [construct_coordinate(Data(x=x[i], y=y[i], world_pos=x_world[i], pos=mesh_pos, cell=cell)) for i in range(x.shape[0])]
        
        return all_data
```

每个样例建Graph， 利用torch_geometric的Data和DataLoader，可以处理每个图里面网格数量不同的情况。

**Comment by SereinQAQ** [comment]
但是在class Grapher(nn.Module):中    
def forward(self, data):
        x = torch.cat([data.x, data.pos], dim=-1)
        x = self.in_mlp(x)
        x_in = x

        for i in range(self.num_layers):
            x = self.blocks[i](x, data.pos, batch=data.batch)

        x_out = self.out_mlp(self.ln(x + x_in))

        return x_out
在训练过程中没有用到world_pos，网络的输入只有data.x, data.pos。如果是动态网格的问题不应该使是用pos预测world_pos嘛

**Comment by lizhihao2022** [comment]
这次任务没有考虑预测pos，而是当作input feature，预测stress。

**Comment by SereinQAQ** [comment]
好的嘞谢谢你
